src/lib/vocab/entity/entityHelpers.ts

```diff
@@ -50,3 +51,24 @@ export const checkEntityPath = (path: string): string | null => {
 	}
 	return null;
 };
+
+/**
+ * Takes an entity with orderedItems in its data
+ * and constructs an ordered array of its items
+ * @param orderedCollection
+ * @param ui
+ * @returns null if invalid, otherwise an array of readable entities
+ */
+export const constructOrderedCollection = (
+	orderedCollection: Entity,
+	ui: Ui,
+): Readable<Array<Readable<Entity>>> | null => {
+	//TODO add recovery logic i.e.
+	//if tere are entities in collection, but not orderedIteam array, append
+	//or if there are ids in the array, but not tied to collection; remove them
+	const {entityById} = ui;
+	if (orderedCollection.data.type !== 'OrderedCollection') return null;
```

Comment:
I think it's more future-proof (like if we add a subtype of `OrderedCollection`) and fits patterns elsewhere to duck type here, so this becomes:

```suggestion
	if (!('orderedItems' in orderedCollection.data)) return null;
```

If that doesn't typecheck, nevermind, but I think it does.

Reply:
I was looking at ActivityStreams, something we haven't been considering is that `type` can actually be either a string or an array of strings in the spec. There's more to think through but duck typing feels correct knowing this.

Reply:
Yeah, that makes sense. Duck typing is still new to me as opposed to explicit type checking, but I'll get there eventually :D